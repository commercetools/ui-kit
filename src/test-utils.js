import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider, addLocaleData } from 'react-intl';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import en from 'react-intl/locale-data/en';
import de from 'react-intl/locale-data/de';
import es from 'react-intl/locale-data/es';
import messages from '../i18n/core.json';

addLocaleData(en);
addLocaleData(de);
addLocaleData(es);

const customRender = (
  node,
  {
    locale = 'en', // react-router
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    ...rtlOptions
  } = {}
) => ({
  ...render(
    <IntlProvider locale={locale} messages={messages}>
      <Router history={history}>{node}</Router>
    </IntlProvider>,
    rtlOptions
  ),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
