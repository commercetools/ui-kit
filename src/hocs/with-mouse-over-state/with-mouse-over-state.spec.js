import React from 'react';
import { render, fireEvent } from '../../test-utils';
import withMouseOverState from './with-mouse-over-state';

const Component = withMouseOverState(props => (
  <div
    data-testid="div"
    onMouseOver={props.handleMouseOver}
    onMouseOut={props.handleMouseOut}
  >
    {props.isMouseOver ? 'mouse over' : 'mouse not over'}
  </div>
));

it('should provide isMouseOver as "false" by default', () => {
  const { getByTestId } = render(<Component />);
  expect(getByTestId('div')).toHaveTextContent('mouse not over');
});

it('should work through the whole cycle', () => {
  const { getByTestId } = render(<Component />);

  // it should provide isMouseOver as "true" when mouse is over
  fireEvent(
    getByTestId('div'),
    new MouseEvent('mouseover', { bubbles: true, cancelable: true })
  );
  expect(getByTestId('div')).toHaveTextContent('mouse over');

  // it should provide isMouseOver as "false" when mouse is up again
  fireEvent(
    getByTestId('div'),
    new MouseEvent('mouseout', { bubbles: true, cancelable: true })
  );
  expect(getByTestId('div')).toHaveTextContent('mouse not over');
});
