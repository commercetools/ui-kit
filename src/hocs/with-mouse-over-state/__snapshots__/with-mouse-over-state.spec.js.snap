// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should output correct tree 1`] = `
<withHandlers(BaseComponent)
  isMouseOver={false}
  setMouseOver={[Function]}
/>
`;
